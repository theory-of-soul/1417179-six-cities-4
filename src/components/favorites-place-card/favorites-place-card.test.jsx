import React from "react";
import renderer from 'react-test-renderer';
import FavoritesPlaceCard from "./favorites-place-card";
import {BrowserRouter} from "react-router-dom";

const place = {
  id: 0,
  mark: `premium`,
  img: `img/apartment-01.jpg`,
  value: 120,
  time: `night`,
  isInBookmark: true,
  rating: 5,
  name: `luxurious apartment at great location`,
  type: `Apartment`
};

describe(`FavoritesPlaceCard component snapshot tests`, () => {
  it(`FavoritesPlaceCard based PlaceCard with special classes and image sizes`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <FavoritesPlaceCard
              place={place}
              onHoverHandler={() => {}}
              addToFavorites={() => {}}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
