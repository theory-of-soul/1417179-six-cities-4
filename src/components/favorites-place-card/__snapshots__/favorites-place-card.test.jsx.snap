// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesPlaceCard component snapshot tests FavoritesPlaceCard based PlaceCard with special classes and image sizes 1`] = `
<article
  className="place-card favorites__card"
  onMouseEnter={[Function]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="place-card__image-wrapper favorites__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="luxurious apartment at great location"
        className="place-card__image"
        height={110}
        src="img/apartment-01.jpg"
        width={150}
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          120
        </b>
        <span
          className="place-card__price-text"
        >
          / 
          night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button place-card__bookmark-button--active"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer/0"
        onClick={[Function]}
      >
        luxurious apartment at great location
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Apartment
    </p>
  </div>
</article>
`;
